src/trivial.rs

```diff
@@ -0,0 +1,301 @@
+//! Trivial transmutation, content-wise
+//!
+//! Functions in this module are guarded from out-of-bounds memory access and
+//! from unsafe transmutation target types through the use of the
+//! [`TriviallyTransmutable`](trait.TriviallyTransmutable.html)) trait.
+//! If a certain type can be safely constructed out of any byte combination,
+//! then it may implement this trait. This is the case for primitive integer
+//! types (e.g. `i32`, `u32`, `i64`), arrays of other trivially transmutable
+//! types, and `repr(C)` structs composed.
+//!
+//! However, they are still not entirely safe because the source data may not
+//! be correctly aligned for reading and writing a value of the target type.
+//! The effects of this range from less performance (e.g. x86) to trapping or
+//! address flooring (e.g. ARM), but this is undefined behavior nonetheless.
+
+
+use self::super::guard::{PermissiveGuard, PedanticGuard, Guard};
+use self::super::base::{transmute_many, from_bytes};
+#[cfg(feature = "std")]
+use self::super::base::transmute_vec;
+use self::super::Error;
+
+
+/// Type that can be constructed from any combination of bytes.
+///
+/// A type `T` implementing this trait means that any arbitrary slice of bytes
+/// of length `size_of::<T>()` can be safely interpreted as a value of that
+/// type with support for unaligned memory access. In most (but not all)
+/// cases this is a [*POD class*](http://eel.is/c++draft/class#10) or a
+/// [*trivially copyable class*](http://eel.is/c++draft/class#6).
+///
+/// This serves as a marker trait for all functions in this module.
+///
+/// *Warning*: if you transmute into a floating-point type you will have a chance to create a signaling NaN,
+/// which, while not illegal, can be unwieldy. Check out [`util::designalise_f{32,64}()`](util/index.html)
+/// for a remedy.
+///
+/// *Nota bene*: `bool` is not `TriviallyTransmutable` because they're restricted to
+/// being `0` or `1`, which means that an additional value check is required.
+///
+/// # Safety
+///
+/// It is only safe to implement `TriviallyTransmutable` for a type `T` if it
+/// is safe for a slice of any arbitrary data `&[u8]` of length `sizeof<T>()`
+/// and _well aligned_ for reads and writes of `T` to be
```

Comment:
This sentence doesn't parse *at all* for me with this part (also asterix italics).

Reply:
I'll try to rephrase this sentence entirely. Let me know of what you think.

Reply:
Mmmm, spicy phrasing!